modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb

```diff
@@ -17,8 +17,8 @@ class UploadsController < ApplicationController
 
       def submit
         Datadog::Tracing.trace('Start IVC File Submission') do
-          Datadog::Tracing.active_trace&.set_tag('form_id', params[:form_number])
           form_id = get_form_id
+          Datadog::Tracing.active_trace&.set_tag('form_id', params[:form_number])
```

Comment:
> I modified the Ruby code to set the tag using the internal form ID instead of the form number.

The only change I see here is moving this line below `form_id = get_form_id`. I guess I need a little more context here.

Reply:
There seems to be a mismatch between how the form ID is being set in the Ruby code and how it's being queried in the Datadog configuration. The problem is that the Ruby code is setting the tag 'form_id' to the form number (e.g., '10-7959F-1'), but it's not converting it to the internal form ID used by the system.
To fix this, I modify the Ruby code to set the tag using the internal form ID instead of the form number.

Reply:
The key change is moving the get_form_id call before setting the Datadog tag, and using the returned form_id value for the tag.

Reply:
@cloudmagic80 😕 I don't see the form id being used here because the name of the tag is `'form_id'` and not the actual `form_id`. I was thinking it would be something like this:
```suggestion
          Datadog::Tracing.active_trace&.set_tag('form_id', form_id)
```

I'm looking at [these docs](https://docs.datadoghq.com/tracing/trace_collection/automatic_instrumentation/dd_libraries/ruby/).